// ytgeturl/src/cjs/ytgeturl.js
'use strict';

const { requestToServer, loadContent } = require('../types/utils.js');
const { convert } = require('../types/convert.js');

const ytgeturl = async (url, lang = 'en') => {
    try {
        const server = await requestToServer(url, lang);
        
        if (!server) {
            throw new Error('Data not found!');
        }
        
        const obj = {};
        const content = loadContent(server);
        
        if (content.mp4) {
            Object.keys(content.mp4).forEach(types => {
                obj.video = {
                    tipo: types,
                    download: convert(content.mp4[types])
                }
            });
        }
        
        if (content.mp3) {
            Object.keys(content.mp3).forEach(types => {
                obj.audio = {
                    tipo: types,
                    download: convert(content.mp3[types])
                }
            });
        }
        
        return {
            sucess: true,
            creator: 'Lm Only',
            result: {
                ...obj
            }
        };
    } catch (e) {
        throw new Error('Erro ao processar dados');
    }
};

module.exports.ytgeturl = ytgeturl;
