// ytgeturl/src/types/utils.js
'use strict';

const { post } = require('axios');
const { load } = require('cheerio');
const { stringify } = require('qs');
const platform = 'youtube';
const retry = 0;
const BASE_URL = 'https://y2mate.bet';

function getAnalyseHost() {
  return `${BASE_URL}/mates/en/analyze/ajax?retry=${retry}&platform=${platform}`;
}

class DownloadURL {
    constructor() {
        this.data = {
            mp4: {},
            mp3: {},
        }
    }
 }

const requestToServer = async (url, lang = 'en') => {
    try {
        const config = stringify({
            url,
            ajax: 1,
            lang
        });

        const {
            data
        } = await post(getAnalyseHost(), config, {
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
            }
        });

        if (data.result && data.status === 'success') {
            return data.result;
        }
        
        return null;
    } catch (e) {
        throw new Error(e);
    }
};
exports.requestToServer = requestToServer;

const loadContent = (html) => {
    try {
        const $ = load(html);
        const downloadURL = new DownloadURL().data;
        
        $('a.btn-success').each((index, element) => {
            const url = $(element).attr('href');
            const type = $(element).data('ftype');
            const quality = $(element).data('fquality') || 'default';
    
            if (url) {
                downloadURL[type][quality] = url;
            }
        });
        
        return downloadURL;
    } catch (e) {
        throw new Error('Falha ao carregar conteúdo HTML');
    }
};
exports.loadContent = loadContent;
